src/granite_orm/migrator.cr

```diff
@@ -0,0 +1,92 @@
+require "./error"
+
+# DB migration tool that prepares a table for the class
+#
+# ```crystal
+# class User < Granite::ORM::Base
+#   adapter mysql
+#   field name : String
+# end
+#
+# User.migrator.drop_and_create
+# # => "DROP TABLE IF EXISTS `users`;"
+# # => "CREATE TABLE `users` (id BIGSERIAL PRIMARY KEY, name VARCHAR(255));"
+#
+# User.migrator(table_options: "ENGINE=InnoDB DEFAULT CHARSET=utf8").up
+# # => "CREATE TABLE ... ENGINE=InnoDB DEFAULT CHARSET=utf8;"
+# ```
+module Granite::ORM::Migrator
+  class Base
+    @quoted_table_name : String
+
+    def initialize(klass, @table_options = "")
+      @quoted_table_name = klass.quoted_table_name
+    end
+    
+    def drop_and_create
+      down
+      up
+    end      
+
+    def down
+    end
+
+    def up
```

Comment:
and this to create?

Reply:
Initially, I imagined cooperating with **micrate** as below, so I choose these names.
```
-- +micrate Up
User.migrator.up

-- +micrate Down
User.migrator.down
```

But, I found that **micrate** can handle only SQL string. Therefore, since there are no constraints now, I think that it is better to change the name as you say. 😺